### PR TITLE
create-lazy-loading

### PR DESCRIPTION
**Pull request by prime1999**
## Changes proposed

-- create a skeleton component to show while data is been fetched
-- extracted the repo, activity, and stats list to another component for more code readability

- [ ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

## Summary by CodeRabbit

- **New Features**
	- Introduced `ActivityList` component to display user activities.
	- Introduced `RepoList` component for rendering repository information.
	- Added `ActivitySkeleton`, `RepoListSkeleton`, and `StatSkeleton` for improved loading states.

- **Enhancements**
	- Implemented dynamic imports for components to optimize loading performance.
	- Reorganized JSX structure for better readability and maintenance.
	- Updated `Stats` component to use dynamic imports and modernized `async/await` syntax.

- **Style**
	- Adjusted code indentation for better clarity and consistency across files.

- **Documentation**
	- Updated `tailwind.config.js` comments for clearer configuration guidance.

### CODE REVIEW

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
It is difficult to review with so many "space" changes, please check your prettier settings

**Comment by prime1999** [comment]
I have resolved the prettier settings

File: tailwind.config.js [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-  theme: {
-    extend: {
-      backgroundImage: {
-        "gradient-radial": "radial-gradient(var(--tw-gradient-stops))",
-        "gradient-conic":
-          "conic-gradient(from 180deg at 50% 50%, var(--tw-gradient-stops))",
-      },
-    },
-  },
-  plugins: [require("daisyui"), require("@tailwindcss/forms")],
```
Comment (eddiejaoude) [inline tailwind.config.js:17]: I think these indentations are not required?

**Comment by eddiejaoude** [comment]
Thank you for making the changes 👍 Could you please resolve the 2 conflicts. If you need any help let me know

**Comment by prime1999** [comment]
yeah, I can't quiet get what the conflict there is

**Comment by eddiejaoude** [comment]
I think there are conflicts because too many files, it might be easier if you select 1 page and create a new PR for that, (we can merge it sooner), then once merged do another PR for the next page

**Comment by prime1999** [comment]
alright will work on it

**Pull request closed without merging**